src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs

```diff
@@ -180,6 +180,13 @@ public WebSocketsTransport(WebSocketOptions options, Channel<byte[]> application
 
                             await ws.SendAsync(new ArraySegment<byte>(buffer), _options.WebSocketMessageType, endOfMessage: true, cancellationToken: CancellationToken.None);
                         }
+                        catch (WebSocketException socketException)
+                        when (socketException.WebSocketErrorCode == WebSocketError.InvalidState)
```

Comment:
I don't like this fix at all. We should optimistically avoid throwing an exception here even though there is a race. Also why don't we look at the actual state instead of looking at the websocket error code?